docs/references/ingress-messages.mdx

```diff
@@ -0,0 +1,82 @@
+---
+keywords: [intermediate, reference, security, ingress message, query call, update call, error handling, idempotency]
+---
+
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+
+# Ingress messages on ICP
+
+<MarkdownChipRow labels={["Intermediate","Reference", "Security"]} />
+
+## Overview
+
+Ingress messages are calls to ICP canisters that are sent by external entities, such as users, usually through an agent like [agent-js](https://github.com/dfinity/agent-js/tree/main) that runs within a web application, or [agent-rs](https://github.com/dfinity/agent-rs/tree/main) in a Rust application. See also the introductory [call overview](/docs/current/developer-docs/smart-contracts/call/overview). 
+
+For example, you can browse what ingress messages can be sent to the ckBTC ledger by browsing through its [API on the dashboard](https://dashboard.internetcomputer.org/canister/mxzaz-hqaaa-aaaar-qaada-cai). The tool also allows to send messages, using agent-js behind the scenes.
+
+In the following, we discuss ICP's ingress message APIs. While these APIs are defined in detail in the [HTTPS interface specification](/docs/current/references/ic-interface-spec#http-interface), we provide a more high level and intuitive overview. In addition to discussing the happy path, we lay a special focus on error handling. That aspect is particularly important as it can be tricky to determine if an ingress message has actually been successfully executed. In the worst case, misinterpreting errors could lead to bugs such as double spending.
+
+## Types of ingress messages
+
+ICP supports several types of calls, as defined in the [HTTPS interface specification](/docs/current/references/ic-interface-spec#http-interface). These include query and update calls. 
+
+## Query Calls
+Queries don't modify a canister's state and are answered and signed by a single replica. Responses are obtained within milliseconds, but they are typically not signed by the subnet, which implies that a malicious replica or boundary node could arbitrarily modify the response. 
```

Comment:
```suggestion
Queries don't modify a canister's state and are answered and signed by a single replica. Responses can be obtained within milliseconds, but they are typically not signed by the subnet, which implies that a malicious replica or boundary node could arbitrarily modify the response. 
```
Maybe hint that we can have pretty slow queries, though maybe I'm overthinking it ;)

Reply:
Adapted as you suggested. 

> Maybe hint that we can have pretty slow queries

Do you mean they could be slow in case complex computation is performed, or in case the system is under heavy load?

Reply:
I suggested "can be" to imply that it's possible that they are processed in milliseconds, but that they can also be slow (in both cases that you mention). So no further changes needed from my side.